### PR TITLE
Move twofactor to models/login

### PR DESCRIPTION
**Pull request by lunny**
As title.

### CODE REVIEW

File: models/login/twofactor.go [hunk 1]
```diff
@@ -48,6 +67,11 @@ func (t *TwoFactor) GenerateScratchToken() (string, error) {
 	return token, nil
 }
 
+// HashToken return the hashable salt
+func HashToken(token, salt string) string {
```
Comment (wxiaoguang) [inline models/login/twofactor.go:71]: Can we just change `hashToken` to `HashToken`?

File: routers/web/user/setting/security_twofa.go [hunk 2]
```diff
@@ -146,15 +146,15 @@ func EnrollTwoFactor(ctx *context.Context) {
 	ctx.Data["Title"] = ctx.Tr("settings")
 	ctx.Data["PageIsSettingsSecurity"] = true
 
-	t, err := models.GetTwoFactorByUID(ctx.User.ID)
+	t, err := login.GetTwoFactorByUID(ctx.User.ID)
 	if t != nil {
 		// already enrolled - we should redirect back!
 		log.Warn("Trying to re-enroll %-v in twofa when already enrolled", ctx.User)
 		ctx.Flash.Error(ctx.Tr("setting.twofa_is_enrolled"))
```
Comment (wxiaoguang) [inline routers/web/user/setting/security_twofa.go:153]: The i18n is `settings.twofa_is_enrolled` ( the `s`). Although I fixed it on a WIP Enforce 2FA branch, but I am afraid that branch won't be merged soon .....

File: routers/web/user/setting/security_twofa.go [hunk 3]
```diff
@@ -172,14 +172,14 @@ func EnrollTwoFactorPost(ctx *context.Context) {
 	ctx.Data["Title"] = ctx.Tr("settings")
 	ctx.Data["PageIsSettingsSecurity"] = true
 
-	t, err := models.GetTwoFactorByUID(ctx.User.ID)
+	t, err := login.GetTwoFactorByUID(ctx.User.ID)
 	if t != nil {
 		// already enrolled
 		ctx.Flash.Error(ctx.Tr("setting.twofa_is_enrolled"))
```
Comment (wxiaoguang) [inline routers/web/user/setting/security_twofa.go:178]: As above

**Review by wxiaoguang — APPROVED** [review]

File: routers/web/user/setting/security_twofa.go (hunk 2 above)
Comment (lunny) [reply routers/web/user/setting/security_twofa.go:153]: another PR #17150

File: routers/web/user/setting/security_twofa.go (hunk 3 above)
Comment (lunny) [reply routers/web/user/setting/security_twofa.go:178]: another PR #17150

**Review by 6543 — APPROVED** [review]

**Pull request merged**